### PR TITLE
Sync upstream changes

### PR DESCRIPTION
**Pull request by lbradstreet**
Changes:
- fixed several merge conflicts in gradle files
- disable RAT check, as we now supply files that are subject to license issues, and the rat check fails if the Apache license is not used
- there was a KStreams test error, that I fixed upstream https://github.com/confluentinc/kafka/pull/191/commits/8e7b57fbcd13edcab280725bb808f232f2e60a1b. This required a second merge from trunk
- fixed a major cause of system test failures in upstream AK. I performed a third merge so this fix would be brought CCS in this sync

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -631,9 +631,27 @@ for ( sv in availableScalaVersions ) {
...
+
+def pkgs = [
+    'clients',
+    'examples',
+    'log4j-appender',
+    'streams',
+    'streams:examples',
+    'streams:streams-scala',
+    'streams:test-utils',
+    'support-metrics-client',
```
Comment (lbradstreet) [inline build.gradle]: merge conflicts here

File: build.gradle [hunk 2]
```diff
@@ -26,16 +26,17 @@ buildscript {
...
-    classpath 'org.owasp:dependency-check-gradle:4.0.2'
-    classpath "com.diffplug.spotless:spotless-plugin-gradle:3.17.0"
-    classpath "gradle.plugin.com.github.spotbugs:spotbugs-gradle-plugin:1.6.9"
+    classpath "org.ajoberstar.grgit:grgit-core:$versions.grgit"
+    classpath "com.github.ben-manes:gradle-versions-plugin:$versions.gradleVersionsPlugin"
+    classpath "org.scoverage:gradle-scoverage:$versions.scoveragePlugin"
+    classpath "com.github.jengelman.gradle.plugins:shadow:$versions.shadowPlugin"
+    classpath "org.owasp:dependency-check-gradle:$versions.owaspDepCheckPlugin"
+    classpath "com.diffplug.spotless:spotless-plugin-gradle:$versions.spotlessPlugin"
+    classpath "gradle.plugin.com.github.spotbugs:spotbugs-gradle-plugin:$versions.spotbugsPlugin"
```
Comment (lbradstreet) [inline build.gradle]: merge conflict here

File: gradle/dependencies.gradle [hunk 3]
```diff
@@ -52,13 +52,19 @@ versions += [
   apacheds: "2.0.0-M24",
   argparse4j: "0.7.0",
   avro: "1.8.1",
-  bcpkix: "1.60",
+  bcpkix: "1.61",
```
Comment (lbradstreet) [inline gradle/dependencies.gradle]: merge conflicts in following lines

**Comment by ijuma** [comment]
What is the source of the conflicts, support metrics changes?

File: build.gradle (hunk 2 above)
Comment (gwenshap) [reply build.gradle]: Its odd that our old dependencies diverged from AK. I wonder when/how that happened.

**Comment by lbradstreet** [comment]
Yes, primarily support-metrics-* changes, and the addition of the gradle plugin `com.commercehub.gradle.plugin:gradle-avro-plugin:0.10.0`

File: build.gradle (hunk 2 above)
Comment (lbradstreet) [reply build.gradle]: Hmm, yes. I'll do another pass to make sure everything is correct.

Comment (lbradstreet) [reply build.gradle]: All looks fine.

Comment (ijuma) [reply build.gradle]: I don't think they diverged. We updated them upstream and that created conflicts due to changes unrelated to versions. Right @lbradstreet?

**Comment by lbradstreet** [comment]
Manually merged.

File: build.gradle (hunk 2 above)
Comment (lbradstreet) [reply build.gradle]: Correct

**Pull request merged**